src/history.ts

```diff
@@ -0,0 +1,92 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import { promisify } from 'node:util';
+
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+
+const filename = 'history.json';
+
+interface ImageInfo {
+  image: string;
+  type: string;
+  location: string;
+}
+
+export class History {
+  infos: ImageInfo[] = [];
+
+  constructor(private readonly storagePath: string) {}
+
+  async loadFile() {
+    // check if history file exists, and load history from previous run
+    try {
+      if (!fs.existsSync(this.storagePath)) {
+        return;
+      }
+
+      const filePath = path.resolve(this.storagePath, filename);
+      if (!fs.existsSync(filePath)) {
+        return;
+      }
+
+      const infoBuffer = await readFile(filePath);
+      this.infos = JSON.parse(infoBuffer.toString('utf8'));
+    } catch (err) {
+      console.error(err);
```

Comment:
don't know if it should be displayed back to the user like with an showErrorMessage ?

Reply:
For now the only user impact is 'we can't remember the path you used the last time you ran Podman Desktop', so I didn't feel like it's necessary to warn the user. If we use it for more critical functions then yes (although I'm also hoping there would be a move to OCI registry at some point).